src/renderer/store/actions.js

```diff
@@ -320,24 +321,28 @@ async function fetchRecentTransactions({ commit }) {
   }
 }
 
-async function fetchTradeHistory({ state, commit }, { marketName, isRequestSilent }) {
+async function fetchTradesBucketed({ commit }, { marketName, interval, from, to }) {
+  try {
+    commit('startRequest', { identifier: 'fetchTradesBucketed' });
+
+    const apiBuckets = await dex.fetchTradesBucketed(marketName, interval, from, to);
+
+    commit('setTradesBucketed', apiBuckets);
+    commit('endRequest', { identifier: 'fetchTradesBucketed' });
+  } catch (message) {
+    alerts.networkException(message);
+    commit('failRequest', { identifier: 'fetchTradesBucketed', message });
+  }
+}
+
+async function fetchTradeHistory({ commit }, { marketName, isRequestSilent }) {
   let history;
   commit(isRequestSilent ? 'startSilentRequest' : 'startRequest',
     { identifier: 'fetchTradeHistory' });
 
   try {
-    let apiBuckets;
-    let promiseFetchTradesBucketed;
-    if (state.tradeHistory && state.tradeHistory.apiBuckets && state.tradeHistory.marketName === marketName) {
-      apiBuckets = state.tradeHistory.apiBuckets;
-    } else {
-      promiseFetchTradesBucketed = dex.fetchTradesBucketed(marketName);
-    }
     history = await dex.fetchTradeHistory(marketName);
```

Comment:
Why did you switch so that these are no longer called in parallel. this one now has to return before the other call will be made?

Reply:
I think this one still applies, but your other fixes look good.